### PR TITLE
Update android external extensions layout

### PR DESCRIPTION
**Pull request by haoxli**
XWALK-5746 Move external extensions management framework down into Crosswalk Webview.

- Enhance build.sh to support crosswalk versions
- Put external extension into app source folder
- Migrate external extensions creation framework from xwalk_app_runtime
  layer to xwalk_core_library

### CODE REVIEW

**Comment by haoxli** [comment]
This PR is needed from crosswalk 18.46.452.0. @zqzhang

File: extensions-android/README.md [hunk 1]
```diff
@@ -33,9 +33,9 @@ Android.
 Pre-condition:
 
   Update crosswalk version in `xwalk-echo-extension-src/build.xml` before run `build.sh`.<br />
-  e.g. change stable 8.37.189.14 to canary 15.44.375.0
+  e.g. change stable 8.37.189.14 to canary 18.46.452.0
```
Comment (zqzhang) [inline extensions-android/README.md]: This is not so good. Can we do this in build.sh automatically?

File: extensions-android/README.md [hunk 2]
```diff
@@ -46,7 +46,7 @@ executable, e.g. on Linux:
 
 Then run it with:
 
-    ./build.sh
+    ./build.sh <version> <arch> <mode>
```
Comment (zqzhang) [inline extensions-android/README.md]: Better support these parameters as optional, e.g. make default values in build.sh.

File: extensions-android/README.md (hunk 1 above)
Comment (haoxli) [inline extensions-android/README.md]: We can change the version code by auto, but it's difficult for canary/beta/stable.

**Comment by zqzhang** [comment]
LGTM, @rakuco

**Pull request merged**